go-controller/pkg/ovn/egressgw.go

```diff
@@ -696,42 +712,104 @@ func (oc *Controller) addHybridRoutePolicyForPod(podIP net.IP, node string) erro
+		}
+
+		// delete hybrid policy to bypass lr-policy in GR, only if there are zero pods on this node.
+		ipv4HashedAS, ipv6HashedAS := as.GetASHashNames()
+		ipv4PodIPs, ipv6PodIPs := as.GetIPs()
+		deletePolicy := false
 		var l3Prefix string
+		var matchSrcAS string
 		if utilnet.IsIPv6(podIP) {
 			l3Prefix = "ip6"
+			if len(ipv6PodIPs) == 0 {
+				deletePolicy = true
+			}
+			matchSrcAS = ipv6HashedAS
 		} else {
 			l3Prefix = "ip4"
+			if len(ipv4PodIPs) == 0 {
+				deletePolicy = true
+			}
+			matchSrcAS = ipv4HashedAS
 		}
-		var matchDst string
-		var clusterL3Prefix string
-		for _, clusterSubnet := range config.Default.ClusterSubnets {
-			if utilnet.IsIPv6CIDR(clusterSubnet.CIDR) {
-				clusterL3Prefix = "ip6"
-			} else {
-				clusterL3Prefix = "ip4"
+		if deletePolicy {
+			var matchDst string
+			var clusterL3Prefix string
+			for _, clusterSubnet := range config.Default.ClusterSubnets {
+				if utilnet.IsIPv6CIDR(clusterSubnet.CIDR) {
+					clusterL3Prefix = "ip6"
+				} else {
+					clusterL3Prefix = "ip4"
+				}
+				if l3Prefix != clusterL3Prefix {
+					continue
+				}
+				matchDst += fmt.Sprintf(" && %s.dst != %s", l3Prefix, clusterSubnet.CIDR)
 			}
-			if l3Prefix != clusterL3Prefix {
-				continue
+			matchStr := fmt.Sprintf(`inport == "%s%s" && %s.src == $%s`, types.RouterToSwitchPrefix, node, l3Prefix, matchSrcAS)
+			matchStr += matchDst
+
+			intPriority, _ := strconv.Atoi(types.HybridOverlayReroutePriority)
+
+			logicalRouter := nbdb.LogicalRouter{}
+			logicalRouterPolicyRes := []nbdb.LogicalRouterPolicy{}
+			opModels := []libovsdbops.OperationModel{
+				{
+					ModelPredicate: func(lrp *nbdb.LogicalRouterPolicy) bool {
+						return lrp.Priority == intPriority && lrp.Match == matchStr
+					},
+					ExistingResult: &logicalRouterPolicyRes,
+					DoAfter: func() {
+						logicalRouter.Policies = libovsdbops.ExtractUUIDsFromModels(&logicalRouterPolicyRes)
+					},
+					BulkOp: true,
+				},
+				{
+					Model:          &logicalRouter,
+					ModelPredicate: func(lr *nbdb.LogicalRouter) bool { return lr.Name == ovntypes.OVNClusterRouter },
+					OnModelMutations: []interface{}{
+						&logicalRouter.Policies,
+					},
+				},
+			}
+			if err := oc.modelClient.Delete(opModels...); err != nil {
+				return fmt.Errorf("failed to remove policy: %s, on: %s, err: %v", matchStr, types.OVNClusterRouter, err)
 			}
-			matchDst += fmt.Sprintf(" && %s.dst != %s", l3Prefix, clusterSubnet.CIDR)
 		}
-		matchStr := fmt.Sprintf(`inport == "%s%s" && %s.src == %s`, types.RouterToSwitchPrefix, node, l3Prefix, podIP)
-		matchStr += matchDst
-
+		if len(ipv4PodIPs) == 0 && len(ipv6PodIPs) == 0 {
+			// delete address set.
+			err := as.Destroy()
+			if err != nil {
+				return fmt.Errorf("failed to remove address set: %s, on: %s, err: %v",
+					as.GetName(), node, err)
+			}
+		}
+	} else if force {
+		klog.Infof("SURYA")
```

Comment:
:)

Reply:
ha good catch, looks like I didn't remove this post testing.